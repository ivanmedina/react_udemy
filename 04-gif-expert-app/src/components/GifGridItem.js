import React from 'react';

export const GrifGridItem=({url, title})=>{
    // console.log(img);
    return(
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}