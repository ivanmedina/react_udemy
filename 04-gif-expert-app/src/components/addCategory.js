import React, { useState } from 'react'
import propTypes from "prop-types";

export function AddCategory({setCategories}) {
    const [inputValue, setInputValue]=useState('');

    const handleInputChange=(e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit=(e)=>{
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategories(categorias=>[ inputValue, ...categorias]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type = "text"
                value = { inputValue }
                onChange = { handleInputChange}
            />
        </form>
        
    )
}

AddCategory.propTypes={
    setCategories: propTypes.func.isRequired
};

export default AddCategory
