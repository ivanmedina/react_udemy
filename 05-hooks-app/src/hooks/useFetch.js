import {useState,useEffect,useRef} from 'react'
export const useFetch = (url) => {


    const [state, setState] = useState({data:null, loading:true, error:null})

    const isMounted=useRef(true);

    useEffect(()=>{
        return ()=>{

            isMounted.current=false;
        }
    },[])

    useEffect(()=>{
        setState({data:null, loading:true, error:null})
        fetch(url).
            then(resp=>resp.json()).
            then(data => {
                setTimeout(() => {

                    if(isMounted.current){
                        setState({
                            loading:false,
                            error:null,
                            data
        
                        })
                    }else{console.log('error evitado')}
                }, 5000);
            })
    },[url]);

    return state;
}
