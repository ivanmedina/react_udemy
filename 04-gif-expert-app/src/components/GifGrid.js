// import React, { useState, useEffect } from "react";
import React from 'react';
import {useFetchGifs} from '../hooks/useFetchGifs';
import { GifGridItem } from "./GifGridItem";
// import {getGifs} from '../helpers/getGifs'

export const GifGrid = ({category})=>{

    const {data,loading}= useFetchGifs(category);
    
    // const [gifs, setGifs]=useState([]);


    return(
        <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>

            {loading && <p className="animate__animated animate__flash">cargando ...</p>}

            <div className="card-grid">
                    {
                        data.map( img =>(
                            <GifGridItem 
                                key={img.id}    
                                { ...img}
                            />
                        ))
                    }
            </div>
        </>
    )

}


