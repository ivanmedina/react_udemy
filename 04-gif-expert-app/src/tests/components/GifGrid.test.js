import React from 'react';
import {shallow} from 'enzyme';
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el <GifGrid />',()=>{
    
    const category='Gamer';

        test('debe de mostrarse correctamente',()=>{
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe cargar items cuando se cargan imagenes useFetchGifs', ()=>{
        const gifs=[{
            id:'ABC',
            url:'https://localhost/elementos/cosa.jpg',
            title: 'Cualquier cosa'
        },{
            id:'123',
            url:'https://localhost/elementos/cosa.jpg',
            title: 'Cualquier cosa'
        },]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wraper=shallow(<GifGrid category={category}/>);
        expect(wraper).toMatchSnapshot();
        
        expect(wraper.find('p').exists()).toBe(false);
        expect(wraper.find('GifGridItem').length).toBe(gifs.length);
    });
})