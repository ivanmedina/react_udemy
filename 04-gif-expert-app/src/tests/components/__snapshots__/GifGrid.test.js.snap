// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe cargar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Gamer
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/elementos/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/elementos/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Gamer
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    cargando ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
